app/models/user.rb

```diff
@@ -1,14 +1,18 @@
-require 'digest/sha1'
-
 class User < ApplicationRecord
+  devise :database_authenticatable,
+         :registerable,
+         :recoverable,
+         :rememberable, 
+         :validatable,
+         :confirmable,
+         :trackable
+
   has_many :test_passages, dependent: :destroy
   has_many :tests, through: :test_passages
   has_many :created_tests, class_name: 'Test', foreign_key: :user_id
 
-  validates :name, :email, presence: true 
-  validates :email, format: /\w+@\w+\.{1}[a-zA-Z]{2,}/, uniqueness: true
-
-  has_secure_password
+  validates :first_name, :email, presence: true 
```

Comment:
```suggestion
  validates :first_name, presence: true 
```
email девайс проверяет